lib/semantics/expression.cc

```diff
@@ -775,15 +771,13 @@ static MaybeExpr AnalyzeExpr(
     return AsMaybeExpr(ConvertToKind<TypeCategory::Integer>(
         *kind, AsExpr(ImpliedDoIndex{n.source})));
   } else if (n.symbol == nullptr) {
-    context.Say(
-        n.source, "TODO INTERNAL: name was not resolved to a symbol"_err_en_US);
+    // error should have been reported in name resolution
```

Comment:
I've been using `CHECK(false);` macros in these "something previous should've caught this already" cases.

Reply:
`CHECK(!"internal error message");` is better.

Reply:
In this case it's not an internal error. The comment is meant to indicate that there is no need to report an error here because it was (hopefully) reported in name resolution.

Maybe it would be clearer just to say "error was reported in name resolution". Which is will be true once name resolution is complete and bug-free.